model/metrics/dotnet/dotnet-http.yaml

```diff
@@ -0,0 +1,222 @@
+        note: |
+          Determined by using the first of the following that applies
+
+          - Host identifier of the [request target](https://www.rfc-editor.org/rfc/rfc9110.html#target.resource)
+            if it's sent in absolute-form
+          - Host identifier of the `Host` header
+      - ref: server.port
+        requirement_level:
+          conditionally_required: If not default (`80` for `http` scheme, `443` for `https`).
+        brief: >
+          Port identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) HTTP request is sent to.
+        note: >
+          When [request target](https://www.rfc-editor.org/rfc/rfc9110.html#target.resource) is absolute URI, `server.port` MUST match
+          URI port identifier, otherwise it MUST match `Host` header port identifier.
+      - ref: server.socket.address
+
+  - id: metric.dotnet.http.client.active_requests
+    type: metric
+    metric_name: http.client.active_requests
+    brief: "Number of active HTTP requests."
+    instrument: updowncounter
+    unit: "{request}"
+    note: Meter name is `System.Net.Http`.
+    attributes:
+      - ref: http.request.method
+        note: |
+          HTTP request method value is one of the "known" methods listed in [RFC9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-methods)
+          and the PATCH method defined in [RFC5789](https://www.rfc-editor.org/rfc/rfc5789.html).
+
+          If the HTTP request method is not known, it sets the `http.request.method` attribute to `_OTHER`.
+
+          It is not possible at the moment to override the list of known HTTP methods.
+      - ref: url.scheme
+        examples: ['http', 'https', 'ftp']
+      - ref: network.protocol.version
+        examples: ["1.1", "2"]
+      - ref: server.address
+        requirement_level: required
+        brief: >
+          Host identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) HTTP request is sent to.
+        note: |
+          Determined by using the first of the following that applies
+
+          - Host identifier of the [request target](https://www.rfc-editor.org/rfc/rfc9110.html#target.resource)
+            if it's sent in absolute-form
+          - Host identifier of the `Host` header
+      - ref: server.port
+        requirement_level:
+          conditionally_required: If not default (`80` for `http` scheme, `443` for `https`).
+        brief: >
+          Port identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) HTTP request is sent to.
+        note: >
+          When [request target](https://www.rfc-editor.org/rfc/rfc9110.html#target.resource) is absolute URI, `server.port` MUST match
+          URI port identifier, otherwise it MUST match `Host` header port identifier.
+
+  - id: metric.dotnet.http.client.request.duration
+    type: metric
+    metric_name: http.client.request.duration
```

Comment:
This metric seems very similar to the generic [`http.client.duration`](https://opentelemetry.io/docs/specs/semconv/http/http-metrics/#metric-httpclientduration) metric.

Actually, some of the metrics listed in this PR seem to be generic (i.e. applicable cross-language, not just to .NET). Wondering whether we should define those as generic metrics rather than .NET specific?

Reply:
My understanding is that the OTel semantic convention `http.client.duration` metric is being renamed to `http.client.request.duration`, independent of .NET's work. The goal wasn't to define a similar metric but rather to implement what the OTel crew said the standard would be.

I think we want these docs to convey two things:
1. There is a set of metrics already defined in a platform neutral way and .NET is declaring the portion that we have implemented in-box thus far.
2. There is another set of metrics that are specific to .NET libraries. In that case .NET is providing both the specification and the implementation.

Unfortunately the markdown generator that produces this text that doesn't really make that distinction clear.

Reply:
There is a text above metric table that explains that it's essentially the same metric with some minor differences.

> This metric follows common [http.client.request.duration](../http/http-metrics.md#metric-httpclientrequestduration) definition. The `server.socket.address` attribute is not available at the instrumentation point and is not reported.

There is also an explanation of what's measured on .NET (the general metric leaves it up to instrumentation and requires to document it).